### PR TITLE
CXP-1318: IsProductBelongToCatalogQuery

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by tseho — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Persistence/Catalog/Product/IsProductBelongingToCatalogQuery.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    ) {
+    }
+
+    public function execute(Catalog $catalog, string $productUuid): bool
+    {
+        $pqb = $this->productQueryBuilderFactory->create([
+            'filters' => $this->getFilters($catalog),
+            'limit' => 1,
+        ]);
+        $pqb->addFilter('id', Operators::EQUALS, $productUuid);
```
[the remark this answers is not in the document]
Comment (tseho) [reply components/catalogs/back/src/Infrastructure/Persistence/Catalog/Product/IsProductBelongingToCatalogQuery.php:30]: moreover, if you remove this line, the integration tests are still passing.

[the remark this answers is not in the document]
Comment (willy-ahva) [reply components/catalogs/back/src/Infrastructure/Persistence/Catalog/Product/IsProductBelongingToCatalogQuery.php:30]: yes, it is!

[the remark this answers is not in the document]
Comment (tseho) [reply components/catalogs/back/src/Infrastructure/Persistence/Catalog/Product/IsProductBelongingToCatalogQuery.php:30]: Willy, I said that `moreover, if you remove this line, the integration tests are still passing.`
I've tested after validating (sorry) but yes, you can break the query AND still having the tests passing, that's a big issue.

[the remark this answers is not in the document]
Comment (willy-ahva) [reply components/catalogs/back/src/Infrastructure/Persistence/Catalog/Product/IsProductBelongingToCatalogQuery.php:30]: Ok, my page was not actualized and I didn't see it. I prioritize the fix. Thanks @tseho

**Pull request merged**